netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -145,18 +142,7 @@ static NettyClientHandler newHandler(Http2Connection connection,
     frameWriter = new Http2OutboundFrameLogger(frameWriter, frameLogger);
 
     StreamBufferingEncoder encoder = new StreamBufferingEncoder(
-        new DefaultHttp2ConnectionEncoder(connection, frameWriter)) {
```

Comment:
Your question was still valid. Before `listener` was _only_ used in the encoder here. However, `lifecycleManager` must be a field because it is called at other times. So instead of using a `final`-based closure here and the field, I just moved it to `FrameListener.onSettingsRead` which seemed cleaner as well.